delphi-checks/src/main/java/au/com/integradev/delphi/checks/RedundantInheritedCheck.java

```diff
@@ -40,11 +46,78 @@ public class RedundantInheritedCheck extends DelphiCheck {
     for (DelphiNode violationNode : findViolations(routine)) {
-      reportIssue(context, violationNode, MESSAGE);
+      context
+          .newIssue()
+          .onNode(violationNode)
+          .withMessage(MESSAGE)
+          .withQuickFixes(getQuickFixes(violationNode))
+          .report();
     }
     return super.visit(routine, context);
   }
 
+  private static List<QuickFix> getQuickFixes(DelphiNode violationNode) {
+    DelphiNode statement = violationNode;
+    DelphiNode parent = violationNode.getParent();
+    while (!(parent instanceof StatementListNode)) {
+      statement = parent;
+      parent = parent.getParent();
+      if (parent == null) {
+        return null;
+      }
+    }
+    StatementListNode statementListNode = (StatementListNode) parent;
+    int statementIndex = statementListNode.getStatements().indexOf(statement);
+
+    int startLine = statement.getBeginLine();
+    int startCol = statement.getBeginColumn();
+    int endLine = statement.getEndLine();
+    int endCol = statement.getEndColumn();
+
+    if (statementListNode.getStatements().size() > statementIndex + 1) {
+      StatementNode statementNode = statementListNode.getStatements().get(statementIndex + 1);
+      endLine = statementNode.getBeginLine();
+      endCol = statementNode.getBeginColumn();
+    } else {
+      if (statementIndex > 0) {
+        StatementNode previousStatement = statementListNode.getStatements().get(statementIndex - 1);
+        FilePosition startPosition = skipTrailingSemicolon(statementListNode, previousStatement);
+        startLine = startPosition.getBeginLine();
+        startCol = startPosition.getBeginColumn();
+      }
+
+      FilePosition endPosition = skipTrailingSemicolon(statementListNode, statement);
+      endLine = endPosition.getEndLine();
+      endCol = endPosition.getEndColumn();
+    }
+
+    return List.of(
+        QuickFix.newFix("Remove the redundant inherited call")
+            .withEdit(
+                QuickFixEdit.delete(FilePosition.from(startLine, startCol, endLine, endCol))));
+  }
+
+  private static boolean isSemicolon(DelphiNode node) {
+    return node.getChildren().isEmpty() && node.getToken().getType() == DelphiTokenType.SEMICOLON;
+  }
+
+  private static FilePosition skipTrailingSemicolon(
+      StatementListNode statementListNode, DelphiNode node) {
+    int col = node.getEndColumn();
+    int line = node.getEndLine();
+    Optional<DelphiNode> optionalTrailingSemicolon =
+        statementListNode.getChildren().stream()
+            .skip(node.getChildIndex() + 1)
+            .takeWhile(RedundantInheritedCheck::isSemicolon)
+            .findFirst();
+    if (optionalTrailingSemicolon.isPresent()) {
+      DelphiNode lastSemiColon = optionalTrailingSemicolon.get();
+      col = lastSemiColon.getEndColumn();
+      line = lastSemiColon.getEndLine();
+    }
+    return FilePosition.from(line, col, line, col);
```

Comment:
We can return the "end node" of the range here instead of a file position.
Then we can simplify the whole function to:
```java
return statementListNode.getChildren().stream()
            .skip(node.getChildIndex() + 1)
            .takeWhile(RedundantInheritedCheck::isSemicolon)
            .findFirst()
            .orElse(node);
```